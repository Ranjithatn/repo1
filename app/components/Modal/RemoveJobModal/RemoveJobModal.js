import React, { Fragment } from "react";
import { connect } from "react-redux";

import Modal from "../Modal";
import { requestHideModal } from "../../../actions/modal";
import { requestRemoveJob } from "../../../actions/jobs";
import Title from "../../Title/Title";
import Button from "../../Button/Button";
import "./RemoveJobModal.scss";

export const RemoveJobModal = ({ requestHideModal, requestRemoveJob, jobId,formatMessage }) => {
  return (
    <Modal
      className="removeJob-modal"
      title={formatMessage({id:"removeJob"})}
      className="removeJob-modal"
      requestHideModal={requestHideModal}
      content={
        <Fragment>
          <Title is="6" text={formatMessage({ id:"Are you sure you want to remove Job ID "}) + jobId + "?"} />
        </Fragment>
      }
      buttons={
        <Fragment>
          <Button id="removeJob-cancel-btn" text={formatMessage({id:"cancel"})} onClick={requestHideModal} />
          <Button id="removeJob-btn" className="is-primary" text={formatMessage({id:"Yes"})} onClick={() => requestRemoveJob(parseInt(jobId))} />
        </Fragment>
      }
    />
  );
};

export default connect(null, {
  requestHideModal,
  requestRemoveJob
})(RemoveJobModal);