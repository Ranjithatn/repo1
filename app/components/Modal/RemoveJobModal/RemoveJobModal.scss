.modal.removeJob-modal footer {
  justify-content: flex-end;
}