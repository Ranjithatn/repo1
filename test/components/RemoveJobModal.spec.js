import React from "react";
import toJson from "enzyme-to-json";
import { RemoveJobModal } from "../../app/components/Modal/RemoveJobModal/RemoveJobModal";
import { shallow, configure, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

let component, tree, instance;

let props = {
  requestHideModal: jest.fn(),
  requestRemoveJob: jest.fn(),
  formatMessage: jest.fn()
}

beforeEach(() => {
  component = mount(
    <RemoveJobModal {...props} />
  );
  tree = toJson(component);
  instance = component.instance();
});

describe("<RemoveJobModal />", () => {
  it("should render without crashing", () => {
    expect(tree).toMatchSnapshot();
  });
  it('should have a cancel button', () => {
    const output = component.find("button#removeJob-cancel-btn").length;
    expect(output).toEqual(1);
  })
  it('should have a button to confirm remove job', () => {
    const output = component.find("button#removeJob-btn").length;
    expect(output).toEqual(1);
  })
  it('should call requestRemoveJob when clicked', () => {
    const output = component.find("button#removeJob-btn").simulate('click');
    expect(props.requestRemoveJob).toHaveBeenCalled();
  })
  it('should call requestHideModal when clicked', () => {
    const output = component.find("button#removeJob-cancel-btn").simulate('click');
    expect(props.requestHideModal).toHaveBeenCalled();
  })
});
