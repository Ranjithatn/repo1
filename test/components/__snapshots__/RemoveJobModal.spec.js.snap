// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RemoveJobModal /> should render without crashing 1`] = `
<Component
  formatMessage={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "id": "removeJob",
          },
        ],
        Array [
          Object {
            "id": "Are you sure you want to remove Job ID ",
          },
        ],
        Array [
          Object {
            "id": "cancel",
          },
        ],
        Array [
          Object {
            "id": "Yes",
          },
        ],
      ],
    }
  }
  requestHideModal={[MockFunction]}
  requestRemoveJob={[MockFunction]}
>
  <Modal
    buttons={
      <UNDEFINED>
        <Button
          id="removeJob-cancel-btn"
          onClick={[MockFunction]}
          text={undefined}
        />
        <Button
          className="is-primary"
          id="removeJob-btn"
          onClick={[Function]}
          text={undefined}
        />
      </UNDEFINED>
    }
    className="removeJob-modal"
    content={
      <UNDEFINED>
        <Title
          as="h1"
          is="6"
          text="NaN?"
        />
      </UNDEFINED>
    }
    requestHideModal={[MockFunction]}
  >
    <div
      className="modal is-active removeJob-modal"
      style={
        Object {
          "zIndex": "50",
        }
      }
    >
      <div
        className="modal-background"
      />
      <div
        className="modal-card"
      >
        <header
          className="modal-card-head"
        >
          <p
            className="modal-card-title"
          />
          <button
            aria-label="close"
            className="delete is-large"
            name="btn-modal-close"
            onClick={[MockFunction]}
          />
        </header>
        <section
          className="modal-card-body"
        >
          <Title
            as="h1"
            is="6"
            text="NaN?"
          >
            <h1
              className="title is-6"
            >
              NaN?
            </h1>
          </Title>
        </section>
        <footer
          className="modal-card-foot"
        >
          <Button
            id="removeJob-cancel-btn"
            onClick={[MockFunction]}
          >
            <button
              className="button"
              disabled={false}
              id="removeJob-cancel-btn"
              onClick={[MockFunction]}
              style={Object {}}
              type="button"
            />
          </Button>
          <Button
            className="is-primary"
            id="removeJob-btn"
            onClick={[Function]}
          >
            <button
              className="button is-primary"
              disabled={false}
              id="removeJob-btn"
              onClick={[Function]}
              style={Object {}}
              type="button"
            />
          </Button>
        </footer>
      </div>
    </div>
  </Modal>
</Component>
`;
